pkgs/development/gnuradio-modules/mkDerivation.nix

```diff
@@ -0,0 +1,21 @@
+{ stdenv
+, unwrapped
+}:
+
+mkDerivation:
+
+args:
+
+# Check if it's supposed to not get built for the current gnuradio version
+if (builtins.hasAttr "disabled" args) && (builtins.elem unwrapped.versionAttr.major args.disabled) then
+let name = args.name or "${args.pname}-${args.version}"; in
+throw "Package ${name} is incompatible with GNURadio ${unwrapped.versionAttr.major}"
+else
+
+let
+  args_ = {
+    enableParallelBuilding = args.enableParallelBuilding or true;
+    nativeBuildInputs = (args.nativeBuildInputs or []) ++ unwrapped.nativeBuildInputs;
+    buildInputs = (args.buildInputs or []) ++ unwrapped.buildInputs ++ [ unwrapped ];
+  };
+in mkDerivation (args // args_)
```

Comment:
This is something I don't feel entirely confident with - adding all of those `buildInputs` unconditionally. Is it too verbose? As part of that, `cmake` is added as well, which influences the default build system stdenv will try to build the derivation with. Maybe there should be a `buildGnuradioPackage` and `buildGnuradioApplication` with different behaviors? 

OTH, the situation with most applications and gnuradio plugins as of currently - they need almost all of those deps and some of them don't even list the deps they need in their readme because they expect them to be present if you have Gnuradio installed.

Reply:
indeed, adding all the inputs of gnuradio to all its packages (essentially a limited form of propagation) seems a bit far. Maybe if there are documented packages that are always required, then they can be explicitly added here in `args`, but just taking in all `(native)buildInputs` is I think not OK. I typically prefer to be correct over pragmatic though.